synthtool/languages/node.py

```diff
@@ -94,6 +97,74 @@ def get_publish_token(package_name: str):
     return package_name.strip("@").replace("/", "-") + "-npm-token"
 
 
+def extract_clients(filePath: Path) -> List[str]:
+    """
+    parse the client name from index.ts file
+
+    Args:
+        filePath: the path of index.ts.
+    Returns:
+        Array of client name string extract from index.ts file.
+    """
+    content = ""
+
+    try:
+        with open(filePath, "r") as fh:
+            content = fh.read()
+    except FileNotFoundError:
+        err_msg = f"Failed to find {filePath}."
+        log.error(err_msg)
+        raise FileNotFoundError(err_msg)
+    return re.findall(r"\{(.*Client)\}", content)
+
+
+def generate_index_ts(versions: List[str], default_version: str) -> bool:
+    """
+    generate src/index.ts to export the client name and versions in the client library.
+
+    Args:
+      versions: the list of versions, like: ['v1', 'v1beta1', ...]
+      default_version: a stable version provided by API producer. It must exist in argument versions.
+    Return:
+      True/False: return true if successfully generate src/index.ts, vice versa.
+    """
+    # sanitizer the input arguments
+    if len(versions) < 1:
+        err_msg = (
+            "List of version can't be empty, it must contain default version at least."
+        )
+        log.error(err_msg)
+        raise AttributeError(err_msg)
+    if default_version not in versions:
+        err_msg = f"Version {versions} must contain default version {default_version}."
+        log.error(err_msg)
+        raise AttributeError(err_msg)
+
+    # compose default version's index.ts file path
+    versioned_index_ts_path = Path("src") / default_version / "index.ts"
+    clients = extract_clients(versioned_index_ts_path)
+    if len(clients) == 0:
```

Comment:
`if not clients:` is the way to please Python programmers.

Reply:
Done.